addons/l10n_it_edi_sdi_coop/models/account_edi_proxy_client.py

```diff
@@ -0,0 +1,94 @@
+from odoo import models, fields
+from .account_edi_proxy_auth import OdooEdiProxyAuth
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives import serialization
+import requests
+import uuid
+import base64
+
+
+class account_edi_proxy_client(models.Model):
+    _name = 'account_edi_proxy_client.user'
+    _description = 'Account EDI proxy user'
+
+    client_id = fields.Char(index=True)
+    company_id = fields.Many2one('res.company', string='Company', required=True,
+        default=lambda self: self.env.company)
+    edi_identification = fields.Char(help="The unique id that identifies this user for on the edi format, typically the vat")
+    private_key = fields.Binary(help="The key to encrypt all the user's data")
+    refresh_token = fields.Char()
+
+    def make_request(self, route, test_mode, params):
+        ''' Make a request to proxy and handle the generic elements of the reponse (errors, new refresh token).
+        '''
+        payload = {
+            'jsonrpc': '2.0',
+            'method': 'call',
+            'params': params,
+            'id': uuid.uuid4().hex,
+        }
+
+        server_url = 'https://iap-services%s.odoo.com' % ('-test' if test_mode else '')
+        response = requests.post(
+            server_url + route,
+            json=payload,
+            headers={'content-type': 'application/json'},
+            auth=OdooEdiProxyAuth(user=self or False)).json()
+
+        if 'result' not in response:
+            raise Exception('Unexpected error in the proxy')
```

Comment:
We are most likely in a cron.  Raising exceptions is not a good idea.  Anyways, even if we just put it in the return or something, we need a catch mechanism or a message_post (_message_log), ...